### PR TITLE
ensure labels are shifted when computing validation loss

### PR DESCRIPTION
**Pull request by syncdoth**
Related discord thread: https://discord.com/channels/799672011265015819/1162768567821930597/1203983815098372096

I believe that the default modeling code of `BartForCausalLM` has a problem.

`BartForCausalLM.forward` has the following:

```
loss = None
if labels is not None:
    labels = labels.to(logits.device)
    loss_fct = CrossEntropyLoss()
    loss = loss_fct(logits.view(-1, self.config.vocab_size), labels.view(-1))
```

i.e., the label is not shifted. This means that with the current code, the model is **not trained to predict the next token, but the current token**. This PR updates `pretraining.validation.compute_losses` to ensure the loss is computed correctly, that is, doing next token prediction.

### CODE REVIEW

**Comment by RusticLuftig** [comment]
syncdoth, thank you so much for submitting this PR! I've submitted it in https://github.com/RaoFoundation/pretraining/pull/66, because this change has a lot of extra commit history I didn't want to pull into dev. I'll be releasing this shortly after some final testing on the release branch.
